src/ports/postgres/modules/utilities/transform_vec_cols.py_in

```diff
@@ -0,0 +1,513 @@
+(14 rows)
+""".format(schema_madlib=schema_madlib)
+        
+        if not message:
+            return summary_string
+        elif message.lower() in ('usage', 'help', '?'):
+            return usage_string
+        elif message.lower() in ('example', 'examples'):
+            return example_string
+        else:
+            return """
+No such option. Use "SELECT {schema_madlib}.vec2cols()" for help.
+        """.format(schema_madlib=schema_madlib)
+
+class cols2vec:
+    def __init__(self):
+        self.get_cols_helper = vec_cols_helper()
+        self.module_name = self.__class__.__name__
+
+    def validate_args(self, source_table, output_table,
+                      list_of_features, list_of_features_to_exclude, cols_to_output):
+        """
+            Function to validate input parameters
+        """
+        input_tbl_valid(source_table, self.module_name)
+        output_tbl_valid(output_table, self.module_name)
+
+        _assert(list_of_features and list_of_features.strip(), "{0}: List of "
+                    "features cannot be empty".format(self.module_name))
+        if list_of_features.strip() != '*':
+            is_var_valid(source_table, list_of_features)
+        
+        if list_of_features_to_exclude:
+            if list_of_features_to_exclude.strip() == "*":
+                plpy.error("{0}: Cannot exclude all columns from being "
+                    "features".format(self.module_name))
+            elif list_of_features.strip() != '*':
+                plpy.info("{0} NOTICE: will exclude given column(s) even though "
+                    "list of features was not *".format(self.module_name))
+
+        is_var_valid(source_table, list_of_features_to_exclude)
+        is_var_valid(source_table, cols_to_output)
+
+    def get_and_validate_feature_types(self, source_table, features_to_nest):
+        """
+            This function will validate and return the appropriate type to cast
+            the final SQL array. Will fail if feature types do not belong to the
+            same group (numeric, text, etc.) or if any type is an array.
+
+            If all features to nest are of the same type, we just return that type
+            and do not cast. Else, if they only constitute integers and smallints,
+            we cast everything to an integer. Else, if they are all part of the same
+            type group, we return the most comprehensive type in that group. Else, throw an error.
+        """
+        all_cols_and_types = get_cols_and_types(source_table)
+        distinct_types = set([col_type[1] for col_type in all_cols_and_types
+            if col_type[0] in features_to_nest])
+        for expr_type in distinct_types:
+            _assert(not is_valid_psql_type(expr_type, ANY_ARRAY),
+                "{0}: Feature columns to nest cannot be of type array"
+                .format(self.module_name))
+        if len(distinct_types) > 1:
+            if distinct_types == {'integer', 'smallint'}:
```

Comment:
Is there a reason why `bigint` is not here? If the distinct types were `{integer, bigint}`, then this function would return back `double` right? It seems to be fine functionality-wise, in the sense that no precision is lost in the output array though.

I created a table named `t1` with the following schema:
```
madlib-pg94=# \d+ t1;
                                                 Table "public.t1"
 Column |       Type       |                    Modifiers                    | Storage | Stats target | Description
--------+------------------+-------------------------------------------------+---------+--------------+-------------
 id     | integer          | not null default nextval('t1_id_seq'::regclass) | plain   |              |
 c1     | integer          |                                                 | plain   |              |
 c2     | bigint           |                                                 | plain   |              |
 c3     | smallint         |                                                 | plain   |              |
 c4     | double precision |                                                 | plain   |              |
```
Calling `select madlib.cols2vec('t1','t1_out','c1,c2');` resulted in an output array of type `double precision[]`. Is that the expected behavior?